lightning/src/ln/channelmanager.rs

```diff
@@ -4030,6 +4030,10 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					// update to. Instead, we simply document in `PaymentForwarded` that this
 					// can happen.
 				}
+
+				// Seems like if the prev hop force closed then we wont be able to get the channel_id from short_channel_id
+				let from_channel_id = channel_state_lock.short_to_id.get(&hop_data.short_channel_id).map(|chan_id| chan_id.clone());
```

Comment:
would be nice to have a way to always get the channel_id even if it has been closed

Reply:
Hmm, we don't keep that (or, really, any) information after a channel closure. In general `ChannelManager` never stores any historical data at all - it provides users with events and users can figure it out if they want to. I think its fine to have a `None` if the channel has been closed, but you could also provide the SCID and let users do the mapping themselves if they want it as a channel id?